src/util.js

```diff
@@ -82,6 +82,19 @@ module.exports.cleanup = function cleanup(config) {
       results = resultSet.filter((result) =>
         actualCaseIds.includes(Number.parseInt(result.case_id, 10)),
       );
+    } else if (options.strictCaseMatching === true) {
+      actualCaseIds = testrail.getCases();
+      results = resultSet.filter((result) =>
```

Comment:
Yes @billylam , but in all cases we want testrail testrun to be updated with what we can and log errors for what it could not rather than not knowing exactly why the testrun was not updated. If you have better suggestion, that would be great.

Reply:
I am not aware why would someone not want testrun updated with valid cases.

Reply:
That's what `strictCaseMatching === false` is for.  Why don't you just use that in your config?

One use case is, CI/CD will pass instead of failing, and can be unnoticeable especially for larger suites.  Thus if a user is deleting cases, or if incorrectly specified cases are added or the suite is just wrong, it would be much slower to address.

Reply:
I would add the logging for both strict and non-strict matching, but not modify the logic itself around strictCaseMatching, instead moving to strictCaseMatching false in your config.